File: 백준/Silver/1926. 그림/그림.js
const fs = require('fs');

let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [nm, ...arr] = input;

const [n,m] = nm.split(' ').map((v) => +v);

const graph = arr.map((v1) => v1.split(' ').map((v2) => +v2));

const dir = [[0, 1], [0, -1], [1, 0], [-1, 0]];

const visited = Array.from(Array(n), () => Array(m).fill(false));

const bfs = (graph, sx, sy) => {
  const q = [];
  q.push([sx, sy]);
  visited[sx][sy] = true;

  let area = 1;

  
  while(q.length !== 0) {
    const [x, y] = q.shift();

    for(let i = 0; i < 4; i++) {
      const nx = x + dir[i][0];
      const ny = y + dir[i][1];

      if(nx < 0 || nx >= n || ny < 0 || ny >= m) {
        continue;
      }

      if(!visited[nx][ny] && graph[nx][ny] === 1) {
        q.push([nx, ny]);
        visited[nx][ny] = true;
        area++;
      }
    }
  }

  return area;
}

let paintCount = 0;
let maxArea = 0;

for(let i = 0; i < n; i++) {
  for(let j = 0; j < m; j++) {
    if(!visited[i][j] && graph[i][j] === 1) {
      paintCount++;
      const area = bfs(graph, i, j);
      maxArea = Math.max(area, maxArea);
    }
  }
}

console.log(paintCount);
console.log(maxArea);

File: 백준/Silver/1926. 그림/그림.py
import sys
from collections import deque

n, m = map(int, sys.stdin.readline().rstrip().split())

array = []

for _ in range(n):
    for _ in range(m):
        line = list(map(int, sys.stdin.readline().rstrip().split()))
        array.append(line)

        
visited = [[0] * m for _ in range(n)]
picture_count = 0
largest_picture_area = 0

dx = [0, 0, 1, -1]
dy = [1, -1, 0, 0]

for i in range(n):
    for j in range(m):

        if array[i][j] == 1 and not visited[i][j]:
            queue = deque()
            queue.append((i, j))
            visited[i][j] = 1
            area = 1

            while queue:
                x, y = queue.popleft()
                for k in range(4):
                    nx = x + dx[k]
                    ny = y + dy[k]

                    if 0 <= nx < n and 0 <= ny < m:
                        if array[nx][ny] == 1 and not visited[nx][ny]:
                            visited[nx][ny] = 1
                            queue.append((nx, ny))
                            area += 1


            largest_picture_area = max(largest_picture_area, area)
            picture_count += 1

print(picture_count)
print(largest_picture_area)
            